openlibrary/catalog/marc/tests/test_parse.py

```diff
@@ -74,13 +74,8 @@ def test_xml(self, i):
 class TestParseMARCBinary:
     @pytest.mark.parametrize('i', bin_samples)
     def test_binary(self, i):
-        expect_filename = "%s/bin_expect/%s" % (test_data, i)
-        data = open("%s/bin_input/%s" % (test_data, i)).read()
-        if len(data) != int(data[:5]):
-            #TODO: Why are we fixing this in test expectations? Investigate.
-            #      affects histoirereligieu05cr_meta.mrc and zweibchersatir01horauoft_meta.mrc
-            data = data.decode('utf-8').encode('raw_unicode_escape')
-        assert len(data) == int(data[:5])
+        expect_filename = '%s/bin_expect/%s' % (test_data, i)
+        data = open('%s/bin_input/%s' % (test_data, i), 'rb').read()
         rec = MarcBinary(data)
```

Comment:
```suggestion
        with open('%s/bin_input/%s' % (test_data, i), 'rb') as in_file:
            rec = MarcBinary(in_file.read())
```
Close dangling file handles.